OpenRA.Mods.Common/Traits/SupportPowers/SelectDirectionalTarget.cs

```diff
@@ -22,32 +22,34 @@ public class SelectDirectionalTarget : IOrderGenerator
 		readonly string order;
 		readonly SupportPowerManager manager;
 		readonly string cursor;
+		readonly string targetPlaceholderCursorPalette;
+		readonly string directionArrowPalette;
 		readonly Animation targetCursor;
 
 		readonly string[] arrows = { "arrow-t", "arrow-tl", "arrow-l", "arrow-bl", "arrow-b", "arrow-br", "arrow-r", "arrow-tr" };
 		readonly Arrow[] directionArrows;
 
 		CPos targetCell;
-		int2 location;
+		int2 targetLocation;
 		int2 dragLocation;
-		bool beginDrag;
+		bool activated;
 		bool dragStarted;
 		Arrow currentArrow;
 
 		public SelectDirectionalTarget(World world, string order, SupportPowerManager manager, string cursor, string targetPlaceholderCursorAnimation,
-			string directionArrowAnimation)
+			string directionArrowAnimation, string targetPlaceholderCursorPalette, string directionArrowPalette)
 		{
 			this.order = order;
 			this.manager = manager;
 			this.cursor = cursor;
+			this.targetPlaceholderCursorPalette = targetPlaceholderCursorPalette;
+			this.directionArrowPalette = directionArrowPalette;
 
 			targetCursor = new Animation(world, targetPlaceholderCursorAnimation);
 			targetCursor.PlayRepeating("cursor");
 
-			for (var i = 0; i < Game.Cursor.Frame; i++)
-				targetCursor.Tick();
-
 			directionArrows = LoadArrows(directionArrowAnimation, world, arrows.Length);
+			Game.HideCursor = true;
```

Comment:
Drawing the whole OG cursor in software can cause noticable regressions in fluidity - drop the frame limiter to something like 10 FPS to see this in action.

It would be better if we could keep this only on mousedown, when the mouse is static anyway.